src/Rules/Methods/IllegalConstructorStaticCallRule.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Methods;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use function in_array;
+
+/**
+ * @implements Rule<Node\Expr\StaticCall>
+ */
+class IllegalConstructorStaticCallRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return Node\Expr\StaticCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node->name instanceof Node\Identifier || $node->name->name !== '__construct') {
+			return [];
+		}
+
+		return $this->isCollectCallingConstructor($node, $scope)
+			? []
+			: ['__construct should not be called outside constructor.'];
```

Comment:
```suggestion
			: ['__construct() should not be called outside constructor.'];
```

Reply:
LGTM

This message is written in other lines, so I changed those.